_11ty/apply-csp.js

```diff
@@ -63,20 +65,35 @@ const addCspHash = async (rawContent, outputPath) => {
       hashes.push.apply(hashes, AUTO_RELOAD_SCRIPTS);
     }
 
-    const csp = dom.window.document.querySelector(
-      "meta[http-equiv='Content-Security-Policy']"
-    );
-    if (!csp) {
-      return content;
-    }
-    csp.setAttribute(
-      "content",
-      csp.getAttribute("content").replace("HASHES", hashes.join(" "))
-    );
-
     content = dom.serialize();
+
+    // write CSP Policy in headers file
+    const headersPath = "./_site/_headers";
```

Comment:
This assumes that this file has already been copied to the output dir, right? This can no longer be assumed in 11ty 1.0. You may need to create it if it doesn't exist yet.

Reply:
Yes. In another project I upgraded to Eleventy 1.0 and I don't encounter any problems. It seems that the command `eleventyConfig.addPassthroughCopy ("_ headers")` in eleventy.js is executed before the code in apply-cps. Do you have any information about this that leads you to think differently? Is it an asynchronous copy problems? In that case, I could remove `eleventyConfig.addPassthroughCopy("_headers")` into eleventy.js and copy the file _headers using the beforeBuild event.

https://www.11ty.dev/docs/events/#beforebuild

Reply:
Using beforeBuild should work. I have definitely encountered that the pass through copy runs concurrently to the build and so while it often works because the copy is fast it doesn't always work.

Reply:
OK. Done